proxy/src/proxy.rs

```diff
@@ -366,7 +375,11 @@ mod tests {
         let (client, server) = tokio::io::duplex(1024);
 
         let (_, server_config) = generate_tls_config("localhost")?;
-        let proxy = tokio::spawn(dummy_proxy(client, Some(server_config), NoAuth));
+        let proxy = tokio::spawn(dummy_proxy(
+            client,
+            Some(TlsConfig::new(server_config)),
```

Comment:
I think we should move `TlsConfig::new(...)` into `generate_tls_config`. This would eliminate the need for such changes.

Reply:
thanks! I didn't notice this.